src/middlewares/isBlocked.js

```diff
@@ -0,0 +1,63 @@
+import models from '../api/models';
+import status from '../helpers/constants/status.codes';
+
+const { User } = models;
+/**
+ * check if user is blocked
+ *
+ * @export
+ * @class checkIfBlocked
+ */
+class checkIfBlocked {
+  /**
+   * this is a middleware which checks if a user is blocked.
+   *
+   * @static
+   * @param {object} req the request
+   * @param {object} res the response
+   * @param { object } next the next route controller to be called
+   * @memberof checkUser
+   * @returns {Object} res
+   */
+  static async checkBlocked(req, res, next) {
+    const { email } = req.params;
+
+    const result = await User.findOne({
+      where: {
+        email,
+        isBlocked: false
+      }
+    });
+
+    return result
+      ? next()
+      : res.status(status.NOT_FOUND).json({ message: `${email} is already blocked` });
+  }
+
+  /**
```

Comment:
@Ramadhan0 no need for checkBlocked and checkUnBlocked. Why do we need two separate methods for this?

Reply:
We first need to check if the user is blocked before we do the job. because we can not block a user who is already blocked 

another one is checking if the user is unblocked because we can not unblock a user who is not blocked. @MemunaHaruna